packages/hash/api/src/graphql/resolvers/taskExecutor/index.ts

```diff
@@ -22,3 +32,217 @@ export const executeDemoTask: Resolver<
     }
   }
 };
+
+export const executeGithubSpecTask: Resolver<
+  Promise<string>,
+  {},
+  GraphQLContext
+> = async (_, __, { dataSources: { taskExecutor } }) => {
+  if (!taskExecutor) {
+    throw new ApolloError(
+      "A task-executor wasn't started, so external tasks can't be started",
+    );
+  } else {
+    try {
+      return await taskExecutor
+        .run_task(Tasks.GithubSpec)
+        .then((res) => JSON.stringify(res));
+    } catch (err: any) {
+      throw new ApolloError(`Task-execution failed: ${err}`);
+    }
+  }
+};
+
+export const executeGithubCheckTask: Resolver<
+  Promise<string>,
+  {},
+  GraphQLContext,
+  MutationExecuteGithubCheckTaskArgs
+> = async (_, { config }, { dataSources: { taskExecutor } }) => {
+  if (!taskExecutor) {
+    throw new ApolloError(
+      "A task-executor wasn't started, so external tasks can't be started",
+    );
+  } else {
+    try {
+      return await taskExecutor
+        .run_task(Tasks.GithubCheck, config)
+        .then((res) => JSON.stringify(res));
+    } catch (err: any) {
+      throw new ApolloError(`Task-execution failed: ${err}`);
+    }
+  }
+};
+
+/** @todo - Make Airbyte types available in api package */
+type AirbyteCatalog = Array<{
+  name?: string;
+  json_schema?: JSONObject;
+}>;
+
+type AirbyteRecords = Array<{
+  /**
+   * the name of this record's stream
+   */
+  stream: string;
+  /**
+   * the record data
+   */
+  data: {
+    [k: string]: unknown;
+  };
+  /**
+   * when the data was emitted from the source. epoch in millisecond.
+   */
+  emitted_at: number;
+  /**
+   * the namespace of this record's stream
+   */
+  namespace?: string;
+  [k: string]: unknown;
+}>;
+
+const ExistingEntityChecker = async (db: DbAdapter, user: User) => {
```

Comment:
This whole object is pretty messy, not sure what we want to do with it. Can merge it with a `@todo` or something but I think generally our API isn't well laid out right now to find an EntityType like we need to here.

It's also based on some assumptions about accountId which I don't think is good.

Reply:
Absolutely, just opened the PR to get it viewed earlier in the process. Agreed on comments if we merge this as is